ui/apps/platform/src/services/AdministrationEventsService.ts

```diff
@@ -86,9 +105,84 @@ export function getAdministrationEvent(id: string): Promise<AdministrationEvent>
         .then((response) => response.data.event);
 }
 
-// TODO ListAdministrationEventsRequest
-export function listAdministrationEvents(): Promise<AdministrationEvent[]> {
+export function listAdministrationEvents(
+    arg: ListAdministrationEventsRequest
+): Promise<AdministrationEvent[]> {
+    const params = qs.stringify(arg, { arrayFormat: 'repeat', allowDots: true });
     return axios
-        .get<ListAdministrationEventsResponse>(eventsUrl)
+        .get<ListAdministrationEventsResponse>(`${eventsUrl}?${params}`)
         .then((response) => response.data.events);
 }
+
+// Given searchFilter from useURLSearch hook, return validated filter argument.
+
+export function getAdministrationEventsFilter(
+    searchFilter: SearchFilter
+): AdministrationEventsFilter {
+    // For consistency with useURLSort hook, especially in case the table columns become sortable,
+    // useURLSearch hook also uses search strings.
+    // See proto/storage/administration_event.proto
+    return {
+        domain: getValue(searchFilter['Event Domain']),
+        level: getLevel(searchFilter['Event Level']),
+        resourceType: getValue(searchFilter['Resource Type']),
+        type: getType(searchFilter['Event Type']),
+    };
+}
+
+type SearchFilterValue = string | string[] | undefined;
+
+function getLevel(arg: SearchFilterValue): AdministrationEventLevel[] | undefined {
+    if (typeof arg === 'string' && isLevel(arg)) {
+        return [arg];
+    }
+
+    if (Array.isArray(arg)) {
+        return arg.filter((item) => isLevel(item)) as AdministrationEventLevel[];
+    }
+
+    return undefined;
+}
+
+function isLevel(arg: string): arg is AdministrationEventLevel {
+    return levels.includes(arg as AdministrationEventLevel);
+}
+
+function getType(arg: SearchFilterValue): AdministrationEventType[] | undefined {
+    if (typeof arg === 'string' && isType(arg)) {
+        return [arg];
+    }
+
+    if (Array.isArray(arg)) {
+        return arg.filter((item) => isType(item)) as AdministrationEventType[];
+    }
+
+    return undefined;
+}
+
+function isType(arg: string): arg is AdministrationEventType {
+    return types.includes(arg as AdministrationEventType);
+}
+
+function getValue(arg: SearchFilterValue): string[] | undefined {
+    if (typeof arg === 'string') {
+        return [arg];
+    }
+
+    if (Array.isArray(arg)) {
+        return arg;
+    }
+
+    return undefined;
+}
+
+// For useURLSort hook.
+
+export const lastOccurredAtField = 'Last Updated';
+export const numOccurrencesField = 'Event Occurrence';
```

Comment:
Given recent discussions on `useURLSort` hook, I think having the valid sort fields in a single location will be beneficial. I believe in many places we are adding the field name to the table header inline.